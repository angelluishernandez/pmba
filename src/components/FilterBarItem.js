import React from "react";
import { FilterBarItemWrapper } from "./styled-components/FilterBarItemWrapper";

const FilterBarItem = ({ genre, handleRadioChange, filtersSelected }) => {
	const isChecked = filtersSelected.includes(genre.toLowerCase());

	return (
		<FilterBarItemWrapper>
			<div>
				<label htmlFor={genre}>{genre}</label>
				<input
					type="radio"
					checked={isChecked}
					value={genre}
					onChange={(e) => handleRadioChange(e)}
				/>
			</div>
		</FilterBarItemWrapper>
	);
};

export default FilterBarItem;
