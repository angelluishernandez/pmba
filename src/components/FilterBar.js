import React, { useState, useEffect } from "react";
import FilterBarItem from "./FilterBarItem";
import { FilterBarWrapper } from "./styled-components/FilterBarWrapper";
import { ButtonContainer } from "./styled-components/ButtonContainer";

import { connect } from "react-redux";
import { filterByGenre } from "../redux/actions/movies.actions";

const FilterBar = ({ filterGenres, setFilterGenres, handleRadioChange }) => {
	useEffect(() => {
		filterByGenre(filterGenres);
	}, [filterGenres]);

	const onReset = () => setFilterGenres([]);

	return (
		<FilterBarWrapper>
			{filterGenres.map((genre, index) => (
				<FilterBarItem
					genre={genre}
					key={index}
					filterGenres={filterGenres}
					handleRadioChange={handleRadioChange}
				/>
			))}

			<ButtonContainer onClick={onReset}>Reset</ButtonContainer>
		</FilterBarWrapper>
	);
};

const mapStateToProps = (state) => ({
	movies: state.movies,
});

const mapDispatchToProps = (dispatch) => ({
	filterByGenre: (genres) => dispatch(filterByGenre(genres)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FilterBar);
