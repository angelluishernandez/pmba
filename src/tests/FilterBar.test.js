import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import React from "react";

import FilterBar from "../components/FilterBar";
import FilterBarItem from "../components/FilterBarItem";

configure({ adapter: new Adapter() });

let genresArr, wrapper;

beforeEach(() => {
	genresArr = ["genreOne", "genreTwo"];
	wrapper = shallow(<FilterBar genresArr={genresArr} />);
});

test("should render FilterBar correctly", () => {
	expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should render one filterBarItem for each genre", () => {
	const filterBarItems = wrapper.find(FilterBarItem);
	expect(filterBarItems).toHaveLength(2);
});
