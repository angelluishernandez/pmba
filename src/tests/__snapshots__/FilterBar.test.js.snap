// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterBar correctly 1`] = `
<styled.div>
  <h4>
    Filter your movies
  </h4>
  <FilterBarItem
    genre="genreOne"
    key="0"
  />
  <FilterBarItem
    genre="genreTwo"
    key="1"
  />
  <div>
    <styled.button>
      Reset
    </styled.button>
  </div>
</styled.div>
`;
